### PR TITLE
Add RSA and ECDSA verification to verifier API

### PR DESCRIPTION
**Pull request by srmalley-sphereon**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
     "bs58": "^4.0.1",
     "compression": "^1.5.2",
     "cors": "^2.7.1",
-    "crypto-ld": "^3.9.0",
+    "crypto-ld": "3.8.0",
```
Comment (nklomp) [inline package.json:37]: Any specific reason why we go down on the dep and fixate it as well?

File: src/lib/credential/index.js [hunk 2]
```diff
@@ -18,17 +23,21 @@ const verifyCredentialStructure = (verifiableCredential) => {
...
         throw new InvalidCredentialStructureError('Credential proof verification method not found.');
     }
 
-    if (!proof.proofPurpose) {
+    if (!proof.proofPurpose && !proof[PROOF_PURPOSE_KEY_EXPANDED]) {
         throw new InvalidCredentialStructureError('Credential proof requires proof purpose field.');
     }
 
-    if (proof.proofPurpose !== 'assertionMethod') {
+    const proofPurpose = proof.proofPurpose || proof[PROOF_PURPOSE_KEY_EXPANDED];
```
Comment (nklomp) [inline src/lib/credential/index.js:34]: The joys of javascript 😸

File: src/lib/verification.js [hunk 3]
```diff
@@ -122,6 +128,61 @@ const _getVerificationFunction = async (vc) => {
...
+    if (typeof verificationMethod === 'object') {
+        verificationMethod = verificationMethod.id;
+    }
+    if (!verificationMethod) {
+        throw new InvalidRequestError('Invalid proof!');
+    }
+    return verificationMethod
+};
+
+const _getRsaSuite = key => {
```
Comment (nklomp) [inline src/lib/verification.js:149]: Guess we could move the suites to a separate file

File: src/lib/verification.js [hunk 4]
```diff
@@ -29,18 +33,20 @@ const verifyCredential = async (verifiableCredential) => {
...
-        });
+        if (Array.isArray(key.type)) {
+            if (key.type.includes('RSAVerificationKey')) {
+                return _getRsaSuite(key);
+            } else {
+                return _getECDSASuite(key);
+            }
+        } else {
+            return _getEd25519Suite(key);
+        }
```
Comment (nklomp) [inline src/lib/verification.js:48]: Can we make a method for getting the correct suite?

**Review by nklomp — COMMENTED** [review]
Some small remarks/clarifications. Nice to see the support for ECDSA and RSA

File: src/lib/verification.js (hunk 3 above)
Comment (srmalley-sphereon) [reply src/lib/verification.js]: yes, updated.

File: package.json (hunk 1 above)
Comment (srmalley-sphereon) [reply package.json:37]: Yeah, the new version of Crypto-ld has a completely new interface and ECDSA suites have not been updated yet to fit that interface. Also RSA and Ed25519 have been removed from crypto ld and set as separate optional drivers, but from what I can tell the drivers aren't ready yet, so best we stick with the old version for now.

File: src/lib/verification.js (hunk 4 above)
Comment (srmalley-sphereon) [reply src/lib/verification.js]: Updated to separate function.

File: src/lib/credential/index.js (hunk 2 above)
Comment (srmalley-sphereon) [reply src/lib/credential/index.js:34]: Haha yes. I also don't quite like the number of constants that are now necessary to handle the expanded RSA proofs, but until there is an RSASignature2021 suite I couldn't find any other options.

File: src/lib/verification.js (hunk 4 above)
Comment (nklomp) [reply src/lib/verification.js]: Nice

**Review by nklomp — APPROVED** [review]

**Pull request merged**